pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -269,108 +284,139 @@ func GetTLSClientForEtcd(tlsConfig *TLSConfig) (*clientv3.Client, error) {
 	if tlsConfig.skipVerify && cfg.TLS != nil {
 		cfg.TLS.InsecureSkipVerify = true
 	}
-
 	return clientv3.New(*cfg)
 }
 
 // applyWatch applies the new watch on etcd and start go routine to periodically take delta snapshots
-func (ssr *Snapshotter) applyWatch() error {
-	client, err := GetTLSClientForEtcd(ssr.tlsConfig)
+func (ssr *Snapshotter) getClientForDeltaSnapshots() error {
+	client, err := GetTLSClientForEtcd(ssr.config.tlsConfig)
 	if err != nil {
 		return &errors.EtcdError{
 			Message: fmt.Sprintf("failed to create etcd client: %v", err),
 		}
 	}
-	ssr.wg.Add(1)
-	go ssr.processWatch(client)
+	ssr.watchClient = client
 	return nil
 }
 
-// processWatch processess watch to take delta snapshot periodically by collecting set of events within period
-func (ssr *Snapshotter) processWatch(client *clientv3.Client) {
+func (ssr *Snapshotter) runDeltaSnapshotter() {
+	ssr.logger.Infof("Running delta snapshots..")
+	ssr.DeltaSnapshotTimer = time.AfterFunc(time.Second*time.Duration(ssr.config.deltaSnapshotIntervalSeconds), ssr.saveDeltaSnapshot)
+	ssr.watchDeltaSnapshots()
+}
+
+func (ssr *Snapshotter) stopDeltaSnapshotter() {
+	ssr.logger.Infof("Terminating delta snapshots.")
+	var stop struct{}
+	ssr.config.SsrStateMutex.Lock()
+	if ssr.DeltaSnapshotTimer == nil {
+		ssr.config.SsrStateMutex.Unlock()
+		return
+	}
+	ssr.config.SsrStateMutex.Unlock()
+	ssr.config.SsrStateMutex.Lock()
+	if ssr.DeltaSnapshotTimer != nil {
+		ssr.DeltaSnapshotTimer.Stop()
+		ssr.DeltaSnapshotTimer = nil
+	}
+	ssr.config.SsrStateMutex.Unlock()
+	ssr.deltaStopCh <- stop
```

Comment:
I am creating the deltaStopSnap channel at snapshotter creation to indicate the delta snapshotter to halt till the next iteration. We need it because we cannot stop the watchCh (it is a receive only channel).  If we are to close the channel and use that as a signal to return from deltaSnapshot function, we will not be passing anything over the channel. We will need to create the channel on each deltasnapshot function invocation. Would that be a good way to do it? Should we use some other mechanism to handle this scenario? I just need to exit the for loop in `watchDeltaSnapshots` without polling overhead.